File: java/parser/src/main/java/gov/lbl/parser/domain/Element_modification_or_replaceable.java
package gov.lbl.parser.domain;

public class Element_modification_or_replaceable {
    private Boolean each;
    private Boolean is_final;
    private Element_modification element_modification;
    private Element_replaceable element_replaceable;

    public Element_modification_or_replaceable(Boolean each, Boolean is_final, Element_modification element_modification, Element_replaceable element_replaceable) {
        this.each = each;
        this.is_final = is_final;
        this.element_modification = element_modification;
        this.element_replaceable = element_replaceable;
    }
}

File: json2mo/elementModificationOrReplaceable.js
function parse (content, rawJson = false) {
  const elementModificationParser = require('./elementModification')
  const elementReplaceableParser = require('./elementReplaceable')

  let moOutput = ''
  if (content.each != null) {
    if (content.each) {
      moOutput += 'each '
    }
  }
  if (rawJson) {
    if (content.is_final != null) {
      if (content.is_final) {
        moOutput += 'final '
      }
    }
  } else {
    if (content.final != null) {
      if (content.final) {
        moOutput += 'final '
      }
    }
  }

  if (content.element_modification != null) {
    moOutput += elementModificationParser.parse(content.element_modification, rawJson)
  } else if (content.element_replaceable != null) {
    moOutput += elementReplaceableParser.parse(content.element_replaceable, rawJson)
  }
  return moOutput
}

module.exports = { parse }
